tsdb/example_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tsdb
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"os"
+	"time"
+
+	"github.com/prometheus/prometheus/model/labels"
+)
+
+func noErr(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+func Example() {
```

Comment:
Since this is in _test.go, we can have it like a proper unit test, without panics, but with assertions